lib/inferno/dsl/fhir_evaluation/rules/all_search_parameters_have_examples.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require_relative '../reference_extractor'
+
+module Inferno
+  module DSL
+    module FHIREvaluation
+      module Rules
+        class AllSearchParametersHaveExamples < Rule
+          attr_accessor :unused_resource_urls
+
+          def check(context)
+            unless ENV.fetch('FHIRPATH_URL')
+              message = 'FHIRPATH_URL is not found. Skipping rule AllSearchParametersHaveExamples.'
+              result = EvaluationResult.new(message, severity: 'warning', rule: self)
+              context.add_result result
+              return
+            end
+
+            @unused_resource_urls = []
+            search_params = context.ig.to_hash[:resources_by_type]['SearchParameter']
```

Comment:
Why is `ig` being converted to a Hash?

Reply:
Addressed https://github.com/inferno-framework/inferno-core/pull/604/commits/62ffdfb34e1bd1a710639da9d6e74186f3aa1678